Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp

```diff
@@ -971,6 +971,34 @@ bool LineLayout::hitTest(const HitTestRequest& request, HitTestResult& result, c
     return false;
 }
 
+void LineLayout::shiftLinesBy(LayoutUnit blockShift)
+{
+    if (!m_inlineContent)
+        return;
+    bool isHorizontalWritingMode = WebCore::isHorizontalWritingMode(flow().style().writingMode());
+
+    for (auto& line : m_inlineContent->displayContent().lines)
+        line.moveInBlockDirection(blockShift, isHorizontalWritingMode);
+
+    for (auto& box : m_inlineContent->displayContent().boxes) {
+        if (isHorizontalWritingMode)
+            box.moveVertically(blockShift);
+        else
+            box.moveHorizontally(blockShift);
+    }
+
+    for (auto& object : m_boxTree.renderers()) {
+        Layout::Box& layoutBox = *object->layoutBox();
```

Comment:
```suggestion
        auto& layoutBox = *object->layoutBox();
```
The variable name is pretty self explanatory -> auto

Reply:
I wish it was, but _everything_ here is representing a layout box. In fact, because this is peeling its way through the integration layers, it's all representing the same layout box, they're all just different types representing the same layout box. Tbh, the fact "layoutBox" is referencing the type Layout::Box was totally lost on me when I was reading this code (because everything here uses `auto`). It's only self-explanatory when you're an expert in this codebase.